### PR TITLE
e2etests: logs and error output modifications

### PR DESCRIPTION
**Pull request by liornoy**
This PR includes 2 commits that modify the e2etest:
1. Add more verbose logs to tests and modify the `wget` error output to include the std message.
2. Modify Gomega error assertions to be aligned with best practices.

### CODE REVIEW

**Comment by liornoy** [comment]
Here is an example of an unreadable failed test output:
```
    May 17 11:52:14.944: FAIL: Timed out after 240.001s.
    Expected
        <*exec.ExitError | 0xc001406140>: {
            ProcessState: {
                pid: 622021,
                status: 32512,
                rusage: {
                    Utime: {Sec: 0, Usec: 170462},
                    Stime: {Sec: 0, Usec: 47018},
                    Maxrss: 223408,
                    Ixrss: 0,
                    Idrss: 0,
                    Isrss: 0,
                    Minflt: 5405,
                    Majflt: 0,
                    Nswap: 0,
                    Inblock: 0,
                    Oublock: 18392,
                    Msgsnd: 0,
                    Msgrcv: 0,
                    Nsignals: 0,
                    Nvcsw: 1074,
                    Nivcsw: 37,
                },
            },
            Stderr: nil,
        }
    to be nil
```

**Comment by liornoy** [comment]
> > Here is an example of an unreadable failed test output:
> > ```
> >     May 17 11:52:14.944: FAIL: Timed out after 240.001s.
> >     Expected
> >         <*exec.ExitError | 0xc001406140>: {
> >             ProcessState: {
> >                 pid: 622021,
> >                 status: 32512,
> >                 rusage: {
> >                     Utime: {Sec: 0, Usec: 170462},
> >                     Stime: {Sec: 0, Usec: 47018},
> >                     Maxrss: 223408,
> >                     Ixrss: 0,
> >                     Idrss: 0,
> >                     Isrss: 0,
> >                     Minflt: 5405,
> >                     Majflt: 0,
> >                     Nswap: 0,
> >                     Inblock: 0,
> >                     Oublock: 18392,
> >                     Msgsnd: 0,
> >                     Msgrcv: 0,
> >                     Nsignals: 0,
> >                     Nvcsw: 1074,
> >                     Nivcsw: 37,
> >                 },
> >             },
> >             Stderr: nil,
> >         }
> >     to be nil
> > ```
> 
> And how would the new version look like?

I'll make an example and put here

**Comment by liornoy** [comment]
```
May 22 13:30:42.892: Timed out after 240.001s.
  timed out waiting to validate service external-local-lb
  Unexpected error:
      <*fmt.wrapError | 0xc000000880>: 
      failed to wget to http://192.168.10.0:80/ from ibgp-single-hop. exit status 1
      {
          msg: "failed to wget to htttp://192.168.10.0:80/ from ibgp-single-hop. exit status 1",
          err: <*exec.ExitError | 0xc000000860>{
              ProcessState: {
                  pid: 165774,
                  status: 256,
                  rusage: {
                      Utime: {Sec: 0, Usec: 40857},
                      Stime: {Sec: 0, Usec: 14095},
                      Maxrss: 55952,
                      Ixrss: 0,
                      Idrss: 0,
                      Isrss: 0,
                      Minflt: 5277,
                      Majflt: 0,
                      Nswap: 0,
                      Inblock: 0,
                      Oublock: 0,
                      Msgsnd: 0,
                      Msgrcv: 0,
                      Nsignals: 0,
                      Nvcsw: 836,
                      Nivcsw: 22,
                  },
              },
              Stderr: nil,
          },
      }
  occurred
  In [It] at: /home/lnoy/go/src/github.com/liornoy/net/metallb/e2etest/bgptests/validate.go:67
```

**Comment by liornoy** [comment]
@fedepaol the simple fact it's telling you that this is a `wget` error and which service and which container was involved will save us time and effort in the future.

**Comment by liornoy** [comment]
I see now that I wrongly thought that switching to the ShouldNot(HaveOccurred()) would supply more details about the err. But anyways it's the recommended way.

**Comment by fedepaol** [comment]
> I see now that I wrongly thought that switching to the ShouldNot(HaveOccurred()) would supply more details about the err. But anyways it's the recommended way.

Can you then change the commit message of the second commit and the PR title? The current one is misleading I think.

**Comment by fedepaol** [comment]
Also, if you want to expose the wget error you can combine the error with the std error produced by the command I think.

**Comment by liornoy** [comment]
Updated example of wget error output:
```
 May 22 18:12:04.158: Timed out after 10.000s.
  timed out waiting to validate service external-local-lb
  Unexpected error:
      <*errors.errorString | 0xc000971860>: 
      failed to wget from ibgp-multi-hop to htttp://192.168.10.0:80/: exit status 1: wget: not an http or ftp url: htttp://192.168.10.0:80/
      
      {
          s: "failed to wget from ibgp-multi-hop to htttp://192.168.10.0:80/: exit status 1: wget: not an http or ftp url: htttp://192.168.10.0:80/\n",
      }
  occurred
  In [It] at: /home/lnoy/go/src/github.com/liornoy/net/metallb/e2etest/bgptests/validate.go:67
```

File: e2etest/bgptests/validate.go [hunk 1]
```diff
@@ -85,7 +86,7 @@ func validateServiceNoWait(svc *corev1.Service, nodes []corev1.Node, c *frrconta
 			address := fmt.Sprintf("http://%s/", hostport)
 			err := wget.Do(address, c)
 			if err != nil {
-				return err
+				return fmt.Errorf("failed to wget from %s to %s: %s", c.Name, address, err)
```
Comment (fedepaol) [inline e2etest/bgptests/validate.go:89]: we should %w errors

Comment (liornoy) [reply e2etest/bgptests/validate.go]: done.

File: e2etest/pkg/wget/wget.go [hunk 2]
```diff
@@ -34,5 +38,8 @@ func Do(address string, exc executor.Executor) error {
 			break
 		}
 	}
-	return err
+	if err != nil {
+		return fmt.Errorf("%s: %s", err.Error(), out)
```
[the remark this answers is not in the document]
Comment (liornoy) [reply e2etest/pkg/wget/wget.go]: done.

**Pull request merged**